src/opm/parser/eclipse/EclipseState/Aquifer/NumericalAquifer/NumericalAquiferConnection.cpp

```diff
@@ -0,0 +1,133 @@
+        return connections;
+    }
+
+    // TODO: we should not need all following the information here
+    using AQUCON = ParserKeywords::AQUCON;
+    NumericalAquiferConnection::NumericalAquiferConnection(const size_t i, const size_t j, const size_t k,
+                                                           const size_t global_index_in, const bool allow_connection_active, const DeckRecord& record)
+    : aquifer_id(record.getItem<AQUCON::ID>().get<int>(0))
+    , I(i)
+    , J(j)
+    , K(k)
+    , global_index(global_index_in)
+    , face_dir(FaceDir::FromString(record.getItem<AQUCON::CONNECT_FACE>().getTrimmedString(0)))
+    , trans_multipler(record.getItem<AQUCON::TRANS_MULT>().get<double>(0))
+    , trans_option(record.getItem<AQUCON::TRANS_OPTION>().get<int>(0))
+    , connect_active_cell(allow_connection_active)
+    , ve_frac_relperm(record.getItem<AQUCON::VEFRAC>().get<double>(0))
+    , ve_frac_cappress(record.getItem<AQUCON::VEFRACP>().get<double>(0))
+    {
+    }
+
+    std::vector<NumericalAquiferConnection> NumericalAquiferConnection::connectionsFromSingleRecord(const EclipseGrid& grid, const DeckRecord& record) {
+        std::vector<NumericalAquiferConnection> cons;
+
+        const size_t i1 = record.getItem<AQUCON::I1>().get<int>(0) - 1;
+        const size_t j1 = record.getItem<AQUCON::J1>().get<int>(0) -1;
+        const size_t k1 = record.getItem<AQUCON::K1>().get<int>(0) - 1;
+        const size_t i2 = record.getItem<AQUCON::I2>().get<int>(0) - 1;
+        const size_t j2 = record.getItem<AQUCON::J2>().get<int>(0) -1;
+        const size_t k2 = record.getItem<AQUCON::K2>().get<int>(0) - 1;
+
+        const std::string str_allow_internal_cells = record.getItem<AQUCON::ALLOW_INTERNAL_CELLS>().getTrimmedString(0);
+        // whether the connection face can connect to active/internal cells
+        // by default NO, which means basically the aquifer should be outside of the reservoir
+        const bool allow_internal_cells = DeckItem::to_bool(str_allow_internal_cells);
+        const FaceDir::DirEnum face_dir
+                = FaceDir::FromString(record.getItem<AQUCON::CONNECT_FACE>().getTrimmedString(0));
+
+        for (size_t k = k1; k <= k2; ++k) {
+            for (size_t j = j1; j <=j2; ++j) {
+                for (size_t i = i1; i <= i2; ++i) {
+                    // TODO: we probably should give a message here
+                    if (!grid.cellActive(i, j, k)) {
+                        continue;
+                    }
+                    /* if (!actnum[grid.getGlobalIndex(i, j, k)]) {
+                        continue;
+                    }*/
+                    if (allow_internal_cells ||
+                        !AquiferHelpers::neighborCellInsideReservoirAndActive(grid, i, j, k, face_dir)) {
+                        const size_t global_index = grid.getGlobalIndex(i, j, k);
+                        cons.emplace_back(NumericalAquiferConnection{i, j, k, global_index, allow_internal_cells, record});
```

Comment:
When you use `emplace_back()` the arguments to `emplace_back()` are forwarded to the constructor - i.e. this should be written:
```
cons.emplace_back( i, j, k, global_index, allow_internal_cells, record );
```